enterprise/middleware.py

```diff
@@ -0,0 +1,28 @@
+from enterprise.models import EnterpriseCustomer, UserDataSharingConsentAudit
+from enterprise.tpa_pipeline import (
+    active_provider_enforces_data_sharing,
+    get_consent_status_for_request,
+    get_enterprise_customer_for_request
+)
+class ResetSessionIfPipelineBrokenMiddleware(object):
+    """
+    Middleware signs the user out if they need to provide data sharing consent,
```

Comment:
Is there some way this can be implemented as a step in the pipeline, rather than as site-wide middleware? Middleware seems like overkill, and the pipeline is where this functionality seems to belong architecturally.

Or, if middleware is required, it would be nicer to move this code to a pipeline step and implement generic middleware in `third_party_auth` which simply ensures that the complete tpa pipeline always runs, and users cannot leave the pipeline prematurely.